src/types/job.rs

```diff
@@ -1,11 +1,61 @@
 //! This module contains types involved with handling phylum processing jobs.
 
+use chrono::{DateTime, Utc};
 use schemars::JsonSchema;
 use serde::{Deserialize, Serialize};
 
 use super::common::*;
 use super::project::*;
-use crate::types::package::{PackageDescriptor, PackageStatus, PackageStatusExtended, PackageType};
+use crate::types::package::{PackageDescriptor, PackageStatusExtended, PackageType};
+
+/// Data shared between the full [`JobResponse`] type and the abbreviated [`JobDescriptor`] type.
+#[derive(PartialEq, PartialOrd, Clone, Debug, Serialize, Deserialize, JsonSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct JobEssentials {
+    pub id: JobId,
+    pub created_at: DateTime<Utc>,
+    pub updated_at: DateTime<Utc>,
+    pub label: Option<String>,
+    pub ecosystem: PackageType,
+    pub project: JobProject,
+    pub score: JobScore,
+    pub num_incomplete: u32,
+}
+
+/// Response type for the API /jobs/<job id> endpoint.
+#[derive(Serialize, Deserialize, JsonSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct JobResponse {
+    #[serde(flatten)]
+    pub essentials: JobEssentials,
+    pub status: Status,
+    pub package_statuses: Vec<PackageStatusExtended>,
+    pub action: Action,
+    pub thresholds: ProjectThresholds,
+}
+
+/// Metadata about a job.
+#[derive(PartialEq, PartialOrd, Clone, Debug, Serialize, Deserialize, JsonSchema)]
+pub struct JobDescriptor {
+    #[serde(flatten)]
+    pub essentials: JobEssentials,
+    pub package_descriptors: Vec<PackageDescriptor>,
+}
+
+#[derive(PartialEq, PartialOrd, Clone, Debug, Serialize, Deserialize, JsonSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct JobProject {
+    pub id: ProjectId,
+    pub name: String,
+}
+
+#[derive(Debug, PartialEq, PartialOrd, Clone, Serialize, Deserialize, JsonSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct JobScore {
+    pub value: f64,
```

Comment:
It would be really helpful to have the per-axis scores for the job included along side the overall job score.